### PR TITLE
B801: add macOS SCI tests for local users and groups

### PR DESCRIPTION
**Pull request by dev-sai-collab**
This PR adds SCI tests for local users and groups under macOS.
It checks:
- If auto logon is enabled/disabled (test name: "AutoLoginStatus")
- If a list of users is presented on the login screen or the user name must be entered manually (test name: "UserNamesShownOnLogin")
- If Apple's insecure default password policy is in use (test name: "DefaultPasswordPolicyUsed")
- If the guest account is enabled/disabled (test name: "GuestAccountEnabled")
- If the currently logged-in user (as in GUI) is a member of the admin group (test name: "CurrentUserAdmin")
- If the root user is enabled/disabled (test name: "RootUserEnabled")

### CODE REVIEW

File: config-dev/Security inventory/macOS/UsersAndGroups/LocalUsersAndGroups.zsh [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    testScore=2
+  fi
+  
+  resultData=$(vlAddResultValue "{}" "Guest Account Enabled" "$guestAccountEnabled")
+  
+  # Create the result object 
+  vlCreateResultObject "$testName" "$testDisplayName" "$testDescription" "$testScore" "$riskScore" "$resultData"
+}
+
+vlCheckCurrentUserAdmin()
```
Comment (martin-kretzschmar) [inline config-dev/Security inventory/macOS/UsersAndGroups/LocalUsersAndGroups.zsh:103]: There is a check for Windows already:
`LocalUsersAndGroups	Local user is admin`

**Review by martin-kretzschmar — CHANGES_REQUESTED** [review]

File: config-dev/Security inventory/Windows/LocalUserAndGroups/LocalUsersAndGroups-User.ps1 [hunk 2]
```diff
@@ -249,7 +249,7 @@ function Get-vlLocalUsersAndGroupsCheck {
       $Output += [PSCustomObject]@{
          Name         = "LUUIsAdmin"
          DisplayName  = "Local user is admin"
-         Description  = "This test determines whether the local user is a member of the local Administrators group."
+         Description  = "Windows: This test determines whether the local user is a member of the local Administrators group.\nmacOS: This test determines if the current user is a member of the group 'admin'."
```
Comment (martin-kretzschmar) [inline config-dev/Security inventory/Windows/LocalUserAndGroups/LocalUsersAndGroups-User.ps1:252]: "\n" does not work in Splunk.

File: config-dev/Security inventory/macOS/LocalUsersAndGroups/LocalUsersAndGroups-User.zsh [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  
+  # Create the result object 
+  vlCreateResultObject "$testName" "$testDisplayName" "$testDescription" "$testScore" "$riskScore" "$resultData"
+}
+
+vlCheckLocalUserIsAdmin()
+{
+  local testName="LUUIsAdmin"
+  local testDisplayName="Local user is admin"
+  local testDescription="Windows: This test determines whether the local user is a member of the local Administrators group.\nmacOS: This test determines if the current user is a member of the group 'admin'."
```
Comment (martin-kretzschmar) [inline config-dev/Security inventory/macOS/LocalUsersAndGroups/LocalUsersAndGroups-User.zsh:30]: "\n" does not work in Splunk.

**Review by martin-kretzschmar — CHANGES_REQUESTED** [review]

File: config-dev/Security inventory/macOS/UsersAndGroups/LocalUsersAndGroups.zsh (hunk 1 above)
Comment (dev-sai-collab) [reply config-dev/Security inventory/macOS/UsersAndGroups/LocalUsersAndGroups.zsh]: Done.

File: config-dev/Security inventory/Windows/LocalUserAndGroups/LocalUsersAndGroups-User.ps1 (hunk 2 above)
Comment (dev-sai-collab) [reply config-dev/Security inventory/Windows/LocalUserAndGroups/LocalUsersAndGroups-User.ps1]: Done.

File: config-dev/Security inventory/macOS/LocalUsersAndGroups/LocalUsersAndGroups-User.zsh (hunk 3 above)
Comment (dev-sai-collab) [reply config-dev/Security inventory/macOS/LocalUsersAndGroups/LocalUsersAndGroups-User.zsh]: Done.

**Pull request merged**